frontend/institution/institutionController.js

```diff
@@ -224,15 +211,26 @@
             }
         };
 
+        function getLegalNature() {
+            InstitutionService.getLegalNatures().then(function success(response) {
+                var legal = _.find(response.data, 
+                    { 'value': institutionCtrl.current_institution.legal_nature });
```

Comment:
To be faster, a better approach is to use the English string being the key and the translation to another language the value, therefore, you can access directly the translation without traverse the entire structure.

You should verify if the backend is serving the string in form of "constants". If not, is better to serve like that to turn generic and portable to any language, using English constant as key to access the translation.